Block/Sales/View.php

```diff
@@ -55,49 +35,58 @@ public function _construct()
-            } else {
-                /** Old data from orders before version 1.6.0 */
-                $html .= __('MyParcel options data not found');
-            }
-        } else {
-            if (is_array($data) && key_exists('date', $data)) {
-                if (key_exists('packageType', $data)) {
-                    $html .= __($data['packageType'] . ' ');
+        $date            = new DateTime($data['date'] ?? '');
+        $dateTime        = $date->format('d-m-Y H:i');
+        $deliveryOptions = DeliveryOptionsAdapterFactory::create((array) $data);
+
+        ob_start();
+
+        if ($deliveryOptions->isPickup()) {
+            try {
+                echo __("{$data['carrier']} location:"), ' ', $dateTime;
+
+                if ($data['deliveryType'] !== 'pickup') {
+                    echo ', ', __($data['deliveryType']);
                 }
 
-                $html .= __('Deliver:') . ' ' . $dateTime;
+                $pickupLocation = $deliveryOptions->getPickupLocation();
 
-                if (key_exists('shipmentOptions', $data)) {
-                    if (key_exists('signature', $data['shipmentOptions']) && $data['shipmentOptions']['signature']) {
-                        $html .= ', ' . __('Signature on receipt');
-                    }
-                    if (key_exists('only_recipient', $data['shipmentOptions']) && $data['shipmentOptions']['only_recipient']) {
-                        $html .= ', ' . __('Home address only');
-                    }
+                if (null !== $pickupLocation) {
+                    echo ', ', $pickupLocation->getLocationName(), ', ', $pickupLocation->getCity(), ' (', $pickupLocation->getPostalCode(), ')';
                 }
+
+            } catch (\Throwable $e) {
+                ObjectManager::getInstance()->get(CheckoutHelper::class)->log($e->getMessage());
+                echo __('MyParcel options data not found');
             }
-        }
+        } elseif (array_key_exists('date', $data)) {
+            if (array_key_exists('packageType', $data)) {
+                echo __($data['packageType']), ' ';
+            }
+
+            echo __('Deliver:'), ' ', $dateTime;
+
+            $shipmentOptions = $deliveryOptions->getShipmentOptions();
 
-        if (is_array($data) && key_exists('browser', $data)) {
-            $html = ' <span title="' . $data['browser'] . '">' . $html . '</span>';
+            if (null !== $shipmentOptions) {
+                if ($shipmentOptions->hasSignature()) {
+                    echo ', ', __('Signature on receipt');
+                }
+                if ($shipmentOptions->hasOnlyRecipient()) {
+                    echo ', ', __('Home address only');
+                }
+            }
         }
 
-        return $html !== false ? '<br>' . $html : '';
+        return htmlentities(ob_get_clean());
```

Comment:
Ik zou er denk ik óf voor kiezen om de html hier te blijven returnen als string, zoals eerst, óf de functienaam aan te passen, want `getCheckoutOptionsHtml()` impliceert natuurlijk niet dat het iets output (dan zou ik `printCheckoutOptions` ervan maken o.i.d.

Als beide niet kunnen, dan in ieder geval het dockblock van deze functie aanpassen om duidelijk te maken wat het nu doet.

Reply:
`htmlentities(ob_get_clean())` is een string. Heb de return type toegevoegd, scherp gezien.